### PR TITLE
[LYN-4618] Upgrade Windows AWSNativeSDK to 1.9.50

### PR DESCRIPTION
**Pull request by onecent1101**
## Details
Windows AWSNativeSDK major version update

New revision:
Add patch to fix c4266 warning. (Tried to disable warning for AWSAuthSigner.h which didn't work, have to path by adding missing override functions)

## Testing
Tested locally: monolithic and non-monolithic build with debug, profile and release different config

Signed-off-by: Liu <liug@amazon.com>

### CODE REVIEW

File: package-system/AWSNativeSDK/FindAWSNativeSDK.cmake.Windows [hunk 1]
```diff
@@ -251,62 +281,21 @@ ly_declare_aws_library(
...
-    NAME 
-        Dependencies
-    BUILD_DEPENDENCIES 
-        3rdParty::AWSNativeSDK::Checksums
-        3rdParty::AWSNativeSDK::Common
-        3rdParty::AWSNativeSDK::EventStream
-)
-
-#### IdentityMetrics ####
-ly_declare_aws_library(
```
Comment (junbo75) [inline package-system/AWSNativeSDK/FindAWSNativeSDK.cmake.Windows:267]: Curious: What's the reason to remove the IdentityMetrics and IdentityLambda libraries? Are they not in use somewhere in the engine?

**Review by junbo75 — APPROVED** [review]
LG

File: package-system/AWSNativeSDK/FindAWSNativeSDK.cmake.Windows (hunk 1 above)
Comment (onecent1101) [reply package-system/AWSNativeSDK/FindAWSNativeSDK.cmake.Windows:267]: correct, they are not used anywhere

File: package-system/AWSNativeSDK/FindAWSNativeSDK.cmake.Windows [hunk 2]
```diff
@@ -53,11 +67,14 @@ function(ly_declare_aws_library)
                     ${AWSNATIVE_SDK_LIB_PATH}/${CMAKE_STATIC_LIBRARY_PREFIX}${ly_declare_aws_library_LIB_FILE}${CMAKE_STATIC_LIBRARY_SUFFIX}
                     ${ly_declare_aws_library_BUILD_DEPENDENCIES}
             )
-
+
```
Comment (lmbr-pip) [inline package-system/AWSNativeSDK/FindAWSNativeSDK.cmake.Windows:70]: nit: whitespace change

**Review by lmbr-pip — APPROVED** [review]
Note: Tested locally does not really convey what you did. Consider explicit statements "Built changes locally in debug, profile and release for Editor, Game and Server Launcher using development branch"

**Review by lawsonamzn — APPROVED** [review]

**Pull request merged**